R/run.R

```diff
@@ -241,6 +256,56 @@ global_setup <- function(lib_path = NULL, cpu_count = NULL, mem_alloc = NULL, te
   script
 }
 
+#' Assemble metadata for a benchmark run
+#'
+#' @param name Benchmark name, i.e. `bm$name`
+#' @param params Named list of parameters for the individual run, i.e. the case
+#' @param cpu_count Number of CPUs allocated
+#' @param n_iter Number of iterations
+#'
+#' @keywords internal
+assemble_metadata <- function(name, params, cpu_count, n_iter) {
+  tags <- params
+  tags[["dataset"]] <- params$source
+  tags[["source"]] <- NULL
+  tags[["cpu_count"]] <- cpu_count
+  tags[["language"]] <- "R"
+
+  arrow_info <- arrow::arrow_info()
+  info <- list(
+    arrow_version = arrow_info$build_info$cpp_version,
+    arrow_compiler_id = arrow_info$build_info$cpp_compiler,
+    arrow_compiler_version = arrow_info$build_info$cpp_compiler_version,
+    benchmark_language_version = version[['version.string']],
+    arrow_version_r = as.character(arrow_info$version)
+  )
+
+  context <- list(
+    arrow_compiler_flags = arrow_info$build_info$cpp_compiler_flags,
+    benchmark_language = "R"
```

Comment:
We might need this for now, but I wonder if this + language above are signs that we should generalize these in the json? We don't have to do that in this PR of course, but a though for when we're designing canonical conbench json

Reply:
Yeah, I think a lot of this is ripe for rearranging, but was just trying to recreate what the Python already generates for the moment. Big picture we're currently slightly constrained by the top-level fields in conbench, behind which I didn't see a lot of reasoning. Didn't see != doesn't exist, but from looking at the conbench code a lot I do think it's worth a conversation if we come up with a well-thought-out alternative, maybe with `language` at/near the top level